src/storage/src/object/mem.rs

```diff
@@ -103,6 +103,14 @@ impl InMemObjectStore {
     }
 }
 
+impl Drop for InMemObjectStore {
+    fn drop(&mut self) {
+        if let Some(sender) = self.compactor_shutdown_sender.lock().take() {
+            let _ = sender.send(());
```

Comment:
better to join the compactor thread (future), otherwise there will be some unexpected errors on SIGINT.

Reply:
Why notify compactor shutdown in objectstore? 
We shall notify compactor shutdown in state-store closed.